### PR TITLE
INFRA-4544 | scuttle init container

### PR DESCRIPTION
**Pull request by wojciechp-remitly**
Initial idea of init container for scuttle.

* docker file for creating image
* change in scuttle to take care of possible infinite loop (shebang)

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+      
+      - name: Build
+        env:
+          CGO_ENABLED: 0
+        run: |
+          go get github.com/cenk/backoff
```
Comment (mszlgr) [inline .github/workflows/tests.yaml:16]: we need to move it up to date with mod :D

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,55 @@
+pipeline {
+    agent {
+        label "default"
+    }
+
+    environment {
+        github = credentials('github-cred-object')
+
+        IMAGE_NAME  = "app/scuttle"
+        REMITLY_ECR = "186612847456.dkr.ecr.us-west-2.amazonaws.com"
```
Comment (mszlgr) [inline Jenkinsfile:10]: On the fence with this change as this repo is public :/ Lets schedule closing this repo as deliver for next sprint

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+all: build release
+
+build:
+	docker build --build-arg="VERSION=${VERSION}" . -f docker/init/Dockerfile -t 186612847456.dkr.ecr.us-west-2.amazonaws.com/app/scuttle:${VERSION}
+
+release:
+	docker push 186612847456.dkr.ecr.us-west-2.amazonaws.com/app/scuttle:${VERSION}
```
Comment (mszlgr) [inline Makefile:7]: how this plan to be used out of jenkins file context that we are sure not to colide with existing version?

File: main.go [hunk 4]
```diff
@@ -56,18 +60,24 @@ func main() {
 		stop := make(chan os.Signal, 2)
 		signal.Notify(stop)
 		for sig := range stop {
-			if proc != nil {
-				proc.Signal(sig)
-			} else {
+			if proc == nil {
+				if syscall.SIGURG == sig {
+					log("Ignoring SIGURG as process didn't started yet")
```
Comment (mszlgr) [inline main.go:65]: ❤️

File: Jenkinsfile (hunk 2 above)
Comment (mszlgr) [reply Jenkinsfile:10]: https://remitly.atlassian.net/browse/INFRA-4687

**Review by mszlgr — APPROVED** [review]

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (wojciechp-remitly) [reply .github/workflows/tests.yaml:16]: most def - once we will have save process with scuttle changes release, I'm planning to merge scuttle to latest version from original repo and work on it

File: Makefile (hunk 3 above)
Comment (wojciechp-remitly) [reply Makefile]: I'll add/extend makefile (and maybe wrap it into op?) with proper command once we make scuttle repo private

File: Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+                    make --version
+                    aws --version
+                '''
+            }
+        }
+
+        stage("Build") {
+            steps {
+                script {
+                    account = sh(returnStdout: true, script: "aws sts get-caller-identity  | jq .Account | tr -d '\"'")
```
Comment (mszlgr) [inline Jenkinsfile:27]:
```suggestion
                    account = sh(returnStdout: true, script: "aws sts get-caller-identity  | jq -r .Account")

**Review by mszlgr — APPROVED** [review]
I'm really excited about this change! Thanks!

**Comment by mszlgr** [comment]
@wojciechp-remitly ! please squash merge to clean up prev commits from history before putting it to main !

**Pull request merged**